PyRDF/Proxy.py

```diff
@@ -50,6 +88,10 @@ def __getattr__(self, attr):
         self._cur_attr = attr  # Stores the name of operation call
         return self._call_handler
 
+    def __del__(self):
+        """Deletes current Proxy and flags the wrapped Node for pruning"""
```

Comment:
`__del__` does not really _delete_ the proxy, but defines the actions to be done right before getting deleted. Also, `for pruning` can be misleading, since there are more conditions. You could put something like:

_Flag the wrapper node when there is no more user references pointing to it, which is one of the conditions to be pruned from the internal graph._